gprofiler/profilers/factory.py

```diff
@@ -30,9 +31,14 @@ def get_profilers(
         if profiler_mode in ("none", "disabled"):
             continue
 
-        if arch not in profiler_config.supported_archs:
-            logger.warning(f"Disabling {profiler_name} because it doesn't support this architecture ({arch})")
-            continue
+        if is_windows():
+            if arch not in profiler_config.supported_windows_archs:
+                logger.warning(f"Disabling {profiler_name} because it doesn't support this architechture ({arch})")
+                continue
+        else:
+            if arch not in profiler_config.supported_archs:
+                logger.warning(f"Disabling {profiler_name} because it doesn't support this architecture ({arch})")
+                continue
```

Comment:
```suggestion
        if arch not in (profiler_config.supported_windows_archs if is_windows() else profiler_config.supported_archs):
            logger.warning(f"Disabling {profiler_name} because it doesn't support this architechture ({arch})")
            continue
```